plugins/inputs/t128_metrics/t128_metrics.go

```diff
@@ -0,0 +1,269 @@
+		for _, permutation := range responseMetric.Permutations {
+			if permutation.Value == nil {
+				continue
+			}
+
+			tags := make(map[string]string)
+			for _, parameter := range permutation.Parameters {
+				tags[parameter.Name] = parameter.Value
+			}
+
+			acc.AddFields(
+				metric.OutMeasurement,
+				map[string]interface{}{metric.OutField: tryNumericConversion(*permutation.Value)},
+				tags,
+				timestamp)
+		}
+	}
+}
+
+func (plugin *T128Metrics) createRequest(baseURL string, metric RequestMetric) (*http.Request, error) {
+	content := struct {
+		Parameters []RequestParameter `json:"parameters,omitempty"`
+	}{
+		metric.Parameters,
+	}
+
+	body, err := json.Marshal(content)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request body for metric '%s': %w", metric.ID, err)
+	}
+
+	request, err := http.NewRequest("POST", fmt.Sprintf("%s%s", baseURL, metric.ID), bytes.NewBuffer(body))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request for metric '%s': %w", metric.ID, err)
+	}
+
+	return request, nil
+}
+
+func configuredMetricsToRequestMetrics(configuredMetrics []ConfiguredMetric) []RequestMetric {
+	requestMetrics := make([]RequestMetric, 0)
+
+	for _, configMetric := range configuredMetrics {
+		for fieldName, fieldPath := range configMetric.Fields {
+			// Sort names for consistency in testing. It's not free, but only happens during startup.
+			parameterNames := make([]string, 0, len(configMetric.Parameters))
+			for parameterName := range configMetric.Parameters {
+				parameterNames = append(parameterNames, parameterName)
+			}
+			sort.Strings(parameterNames)
+
+			parameters := make([]RequestParameter, 0, len(configMetric.Parameters))
+			for _, parameterName := range parameterNames {
+				values := configMetric.Parameters[parameterName]
+				parameters = append(parameters, RequestParameter{
+					Name:    parameterName,
+					Values:  values,
+					Itemize: true,
+				})
+			}
+
+			requestMetrics = append(requestMetrics, RequestMetric{
+				ID:             fieldPath,
+				Parameters:     parameters,
+				OutMeasurement: configMetric.Name,
+				OutField:       fieldName,
+			})
+		}
+	}
+
+	return requestMetrics
+}
+
+func tryNumericConversion(value string) interface{} {
+	if i, err := strconv.Atoi(value); err == nil {
+		return i
+	} else if f, err := strconv.ParseFloat(value, 64); err == nil {
+		return f
+	} else {
+		return value
+	}
+}
+
+type requestLimiter struct {
+	synchronizer chan struct{}
+}
+
+func newRequestLimiter(limit int) *requestLimiter {
+	limiter := &requestLimiter{make(chan struct{}, limit)}
+
+	for i := 0; i < limit; i++ {
+		limiter.done()
+	}
+
+	return limiter
+}
+
+func (l *requestLimiter) wait() {
+	<-l.synchronizer
+}
+
+func (l *requestLimiter) done() {
+	l.synchronizer <- struct{}{}
+}
+
+func init() {
+	inputs.Add("t128_metrics", func() telegraf.Input {
+		return &T128Metrics{
+			Timeout:                 internal.Duration{Duration: time.Second * 5},
```

Comment:
consts for these defaults to make it more clear that this is doing the default prepopulation? It originally confused me what this bit of code was doing